src/org/rascalmpl/dap/RascalDebugAdapter.java

```diff
@@ -506,48 +521,221 @@ public CompletableFuture<Void> pause(PauseArguments args) {
         }, ownExecutor);
     }
 
+    private void outputErrorMessage(String message) {
+        OutputEventArguments errorOutput = new OutputEventArguments();
+        errorOutput.setCategory(OutputEventArgumentsCategory.STDERR);
+        errorOutput.setOutput(message);
+        client.output(errorOutput);
+    }
+
+    private org.rascalmpl.debug.IRascalRuntimeEvaluation.EvalResult evaluateExpression(String expression, int frameId) {
+        assert suspendedState.isSuspended(); // For race condition reason, call to evaluator is only allowrd on suspended state
```

Comment:
asserttions are disabled when we're running rascal. so should we maybe turn this into an `if` with a `IllegalStateException` or something?